vocode/streaming/synthesizer/azure_synthesizer.py

```diff
@@ -358,13 +359,19 @@ def get_message_up_to(
         word_boundary_event_pool: WordBoundaryEventPool,
     ) -> str:
         events = word_boundary_event_pool.get_events_sorted()
-        # for event in events:
-        #     if event["audio_offset"] > seconds:
-        #         ssml_fragment = ssml[: event["text_offset"]]
-        #         # TODO: this is a little hacky, but it works for now
-        #         return ssml_fragment.split(">")[-1]
+        for event in events:
+            if event["audio_offset"] > seconds:
+                ssml_fragment = ssml[: event["text_offset"]]
+                return ssml_fragment.split(">")[-1]
```

Comment:
i don't know what typical SSML looks like well enough to judge it. But the deleted comment says it's hacky, and if this produces an incorrect transcript in even 5% of cases it's a SEV

I was thinking to suggest a try/catch in case `ssml_fragment` doesn't contain `>`. But that would fail to catch a lot of possible problems. Another option is to alert/log if this value is not a substring of `message`. Open to ideas

Reply:
This is what these look like:

```
app-1    | DEBUG:telephony_app.main:[CYRh9ShXLPIR_BsAb_8wiA] ...
Event: {'text': 'on', 'text_offset': 273, 'audio_offset': 6.194375, 'boundary_type': <SpeechSynthesisBoundaryType.Word: 0>}
SSML: <speak xmlns="https://www.w3.org/2001/10/synthesis" version="1.0" xml:lang="en-US"><voice name="en-US-AvaNeural"><prosody pitch="0%" rate="6.0%" volume="-1%">The online search tool allows agents to access up-to-date and detailed information from publicly available sources on the internet to address user requests requiring specificity or accuracy beyond pre-trained knowledge.<break time="100ms" /></prosody></voice></speak>
ssml_fragment: <speak xmlns="https://www.w3.org/2001/10/synthesis" version="1.0" xml:lang="en-US"><voice name="en-US-AvaNeural"><prosody pitch="0%" rate="6.0%" volume="-1%">The online search tool allows agents to access up-to-date and detailed information from publicly available sources
ssml_fragment_split: ['<speak xmlns="https://www.w3.org/2001/10/synthesis" version="1.0" xml:lang="en-US"', '<voice name="en-US-AvaNeural"', '<prosody pitch="0%" rate="6.0%" volume="-1%"', 'The online search tool allows agents to access up-to-date and detailed information from publicly available sources ']
```

<details><summary>Modified getmsgupto with logs</summary>
<p>

```
    def get_message_up_to(
        self,
        message: str,
        ssml: str,
        seconds: float,
        word_boundary_event_pool: WordBoundaryEventPool,
    ) -> str:
        events = word_boundary_event_pool.get_events_sorted()

        for event in events:
            if event["audio_offset"] > seconds:
                self.logger.debug(f"Event: {event}")
                self.logger.debug(f"SSML: {ssml}")
                self.logger.debug(f"ssml_fragment: {ssml[: event['text_offset']]}")
                self.logger.debug(
                    f"ssml_fragment_split: {ssml[: event['text_offset']].split('>')}"
                )
                ssml_fragment = ssml[: event["text_offset"]]
                return ssml_fragment.split(">")[-1]
        return message
```

</p>
</details> 

The ssml will always have `>`. Its hacky because theres definitely a better way to do it and if there were a `>` in the response it would be a problem. 

A few solutions of the top of my head are:

1. Record the char length of the ssml tag. Extract via that. 
`ssml_fragment = ssml[: event["text_offset"]]` -> `ssml[ssml_char_length:event['text_offset']]}`

2. Since we explicitly define the ssml tags in the create_ssml function, we could just pass the leading ssml tags into `get_message_up_to`. OR we could just extract them using the message_text and the ssml. 

<details><summary>ssml leading tag</summary>
<p>

```
<speak xmlns="https://www.w3.org/2001/10/synthesis" version="1.0" xml:lang="en-US"><voice name="en-US-AvaNeural"><prosody pitch="0%" rate="6.0%" volume="-1%">
```

</p>
</details>

Reply:
both your suggestions are on the right track, but seem like variations of the same thing: integrate this with `create_ssml`

I like that idea, but I'm not sure it's feasible to do in this PR. So unless you have an idea that's easy enough to slip into this PR, I recommend doing the substring check cuz it should be easy. Not very useful tbh, but easy and better than silent failure